src/hlcand.c

```diff
@@ -326,7 +332,36 @@ int main(int argc, char *argv[])
 		exit(EXIT_FAILURE);
 	}
 	
+	/* retrieve the name of the created CAN netdevice */
+	if (ioctl(fd, SIOCGIFNAME, ifr.ifr_name) < 0) {
```

Comment:
This is not really working and the application always exits in line `338` no matter if a name was given or not. 
You also don't need it as `ifr.ifr_name` already contains `can0` (for example), it's also logged in `342`.
If the `if` is removed the application will exit in `342` with `invalid ioctl for device`. 

Please do not blindly copy from https://github.com/linux-can/can-utils/blob/master/slcan_attach.c#L225 and test your changes. 
Furthermore this change introduces a kernel freeze in some scenarios I haven't debugged. 

Let me know if you need further help to implement this feature.